// Copyright (c) 2012, Cloudera, inc.

package kudu.cfile;

enum DataType {
  UINT32 = 0;
  STRING = 1;
}

enum EncodingType {
  PREFIX = 0;
  GROUP_VARINT = 1;
}


message CFileHeaderPB {
  required int32 major_version = 1;
  required int32 minor_version = 2;
}

message BlockPointerPB {
  required int64 offset = 1;
  required int32 size = 2;
}

message BTreeInfoPB {
  required BlockPointerPB root_block = 1;
}

message IndexBlockTrailerPB {
  required int32 num_entries = 1;

  enum BlockType {
    LEAF = 0;
    INTERNAL = 1;
  };
  required BlockType type = 2;
}
// TODO: name all the PBs with *PB convention

message CFileFooterPB {
  required DataType data_type = 1;
  required EncodingType encoding = 2;

  required BTreeInfoPB posidx_info = 3;
  optional BTreeInfoPB validx_info = 4;
}